### ISSUE TITLE
resume pipeline from an intemediate step 

### ISSUE THREAD

**Issue by andreagalle** [Toontown-Event-Horizon/panda-utils#2]
I am trying to run the pipeline in two steps, basically splitting the following command:

    python -m panda_utils.assetpipeline input/asset-folder phase_1/models blend2bam bam2egg group_rename[] collide[] egg2bam

into the following two:

    python -m panda_utils.assetpipeline input/asset-folder phase_1/models preblend blend2bam bam2egg
    python -m panda_utils.assetpipeline input/asset-folder phase_1/models group_rename[] collide[] egg2bam

since I want to run (in the middle) an .egg parser to find out all the groups an populate a `model-config.yml` file accordingly, but I can't, since the second command (the second pipeline in a row) basically first of all empty the intermediate folder and nothing remain to start from.

**Comment by Wizzerinus** [comment]
Right now the recommended way to insert steps in the middle of pipeline is to use egg scripting: [documentation here](https://panda-utils.readthedocs.io/en/latest/asset-pipeline/scripting.html). Python scripts are used because they're 1) easiest to be made crossplatform and 2) 95% of the custom steps are eggtree manipulations (although there are some other possibilities like image compression). Something like (a bit of pseudocode)

```py
def run(ctx):
    ctx.cache_eggs()
    for eggtree in ctx.eggs.values():
        for building in eggtree.findall("Group"):
            if "Building" not in building.node_name:
                continue
            
            collide = make_collide_node()
            building.children.insert(0, collide)
```

(do note that this version is not very smart because in your case it will likely insert two collide nodes on each building, since you have a nested mesh structure with the same names. probably an obj importer specific structure, we usually use fbx to import assets from other software.)

I however could add an argument (something like `--keep-intermediate`) to not empty the intermediate folder. I do not know how useful it would be with the existance of scripts, but could be.

Do note that fnmatch should be implemented as of 1.6b4 so you don't need that thing for your usecase (see: [test case](https://github.com/Toontown-Event-Horizon/panda-utils/blob/master/tests/pipeline.py#L37))

**Comment by Wizzerinus** [comment]
Closing as scripts is the intended way of customizing the pipeline, and the intermediate folder shouldn't be relied on (except for debugging).
